### ISSUE TITLE
set fp32 for MatMul failure of TensorRT 8.6.1.1 when converting onnx to trt with version 8.6.1.1

### ISSUE THREAD

**Issue by RobinsonKO** [NVIDIA/TensorRT#4050]
## Description
onnx: https://drive.google.com/file/d/1JgwgwIl71BnJRw2e9FtgV0DGSGzLy0OZ/view?usp=sharing

I  tried to use fp32 for MatMul in self-attention and cross-attention layer on A100, but it seems not work. 
- I inserted Cast layer around MatMul op. Did I use it wrong?

![input](https://github.com/user-attachments/assets/cfb634a6-75dd-44eb-976e-792e05ec19e5)
![output](https://github.com/user-attachments/assets/3251de54-0d3b-4332-b69d-403382bd744a)

In A100, reform the softmax position of onnx's self-attention into output, which can be aligned with the torch KPI, but the same method is not aligned with the torch KPI on orin. （orin KPI dropped by 30%）
- On orin, the softmax result of the last layer of cross-attention needs to be set as the output to align with the KPI of the torch. this is so weird? 
![O](https://github.com/user-attachments/assets/913a7598-f9eb-4734-9853-250d9e854711)

## Environment 

**A100**
**TensorRT Version**: 8.6.1.1
**NVIDIA GPU**: A100
**NVIDIA Driver Version**: 470.199.02
**CUDA Version**: 11.4
Operating System: Linux
Python Version (if applicable): 3.8.10
PyTorch Version (if applicable): 1.10.0

**Orin**
**TensorRT Version**: 8.6.1.1

```[tasklist]
### Tasks
```

**Comment by nvluxiaoz** [comment]
Have you tried to use the ToT version? (e.g. 10.2) I think because you are using weakly_typed networks and I assume you enabled fp16 flags, you will need to set the precision for the specific layer and set a TRT flag to obey precision to force the layers that you want to be executed in fp32. 

In addition, it is possible that in Orin, MHA is not fused into a single kernel, which might also affect perf and accuracy, can you try collect some nsys profiles on both platforms to see if there are any kernel diffs?

**Comment by Wuqiman** [comment]
![image](https://github.com/user-attachments/assets/a5b4d166-7cac-467d-a9ee-10609bb6e3e0)
For MHA precision, we want Matmul cast to fp32，which used to fp16. However, it doesn't work.

**Comment by lix19937** [comment]
>  For_ MHA precision, we want Matmul cast to fp32，which used to fp16. However, it doesn't work.     

Try to mark this matmul as out, then rebuild with trtexec.

**Comment by kevinch-nv** [comment]
You can use the builder flag kSTRONGLY_TYPED to force TensorRT to respect the casts. See https://github.com/NVIDIA/TensorRT/blob/release/10.8/include/NvInfer.h#L10299 for more info.

**Comment by poweiw** [comment]
Closing due to inactive. Please feel free to reopen!
